### ISSUE TITLE
Wrong template order in Clusterization.ipp

### ISSUE THREAD

**Issue by gagnonlg** [acts-project/acts#1825]
In createClusters, the labelClusters function is called like so: 
```
labelClusters<CellCollection, Connect, GridDim>(cells, connect);
```
However, the labelClusters function is defined like so:
```
template <typename CellCollection, size_t GridDim = 2,
          typename Connect =
              DefaultConnect<typename CellCollection::value_type, GridDim>>
void labelClusters(CellCollection& cells, Connect connect = Connect());
```

Therefore the order is wrong in the createClusters wrapper. However, createClusters is called in the unit tests which compile and run, so I need to investigate what is happening, and if there's at all a bug lurking here.

**Comment by gagnonlg** [comment]
As noted by @CarloVarni there was another ordering mistake in the .ipp where the "real" labelClusters is defined. The two mistakes aligned which is why this wasn't caught by the compiler.

**Comment by paulgessinger** [comment]
Does this not get compiled anywhere? I thought we had a unit test cpp where the template is instantiated?

**Comment by gagnonlg** [comment]
Yes it gets compiled in the tests. But, the `createClusters` wrapper *and* the definition of `labelClusters` in the .ipp both had the wrong ordering of template parameters, so it compiled anyways. I intended to have the GridDim parameter before the Connect type because the latter can be derived based on the former, which is what was fixed here.

**Comment by gagnonlg** [comment]
Fixed by #1827
